agents/tests/test_dns_agent.py

```diff
@@ -0,0 +1,110 @@
+# Command to run on VM: python -m pytest agents/tests/test_dns_agent.py
+import os
+import tempfile
+
+import pytest
+from mock import MagicMock, patch
+
+from agents.dns.dns import DNSAgent
+
+
+@pytest.yield_fixture
+def dns_agent():
+    """
+    Create and configure a DNSAgent instance with logging for use in tests.
+    Cleans up the temporary log file after the test completes.
+    """
+    os.environ['PORT'] = '53'
+    logfile = tempfile.NamedTemporaryFile(delete=False)
+    logfile.close()
+
+    agent = DNSAgent()
+    agent.setup_logging(logfile.name)
+
+    yield agent
+
+    if os.path.exists(logfile.name):
+        os.remove(logfile.name)
+
+
+def test_status_to_dict_keys(dns_agent):
+    """
+    Verify that status_to_dict()
+    returns all expected keys in the status dictionary.
+    """
+    dns_agent.collect_server_metadata()
+    result = dns_agent.status_to_dict()
+    required_keys = set(
+        [
+            'os',
+            'hostname',
+            'ip',
+            'server_name',
+            'uptime',
+            'timestamp',
+            'healthy',
+        ]
+    )
```

Comment:
If you want more control, you could mock `.collect_server_metadata()`  method instead of relying on implementation internals. This avoids dependency on metadata collection.